x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizer.java

```diff
@@ -91,7 +90,7 @@ static class ProjectAwayColumns extends Rule<PhysicalPlan, PhysicalPlan> {
         public PhysicalPlan apply(PhysicalPlan plan) {
             var projectAll = new Holder<>(TRUE);
             var keepCollecting = new Holder<>(TRUE);
-            var attributes = new LinkedHashSet<Attribute>();
```

Comment:
This small change took a while to figure out due to a failure in the CSV tests (yey for testing):

`eval nil = concat("", null) | eval is_in = nil in ("Accountant", "Internship", null)`

With separate evals, the reference to the Alias had the exact same properties.
However when the eval combination kicked in, one of the rules caused a new Alias to be created with the same properties however its attribute had _slightly_ different properties - the nullability was true in this case.
This caused the ReferenceAttribute comparison to fail, which caused the test to fail miserably.
By moving to an AttributeSet, the semantic equality is used, regardless of the reference properties.

One could argue that the ReferenceAttribute should NOT copy the properties since the source itself can be manipulated - however that's work for another PR since it's likely going to impact SQL as well.